packages/backend/src/dbt/dbtRpcClient.ts

```diff
@@ -3,64 +3,15 @@ import fetch from 'node-fetch';
 import { DbtCatalog, DbtNode } from 'common';
 import { DbtError, NetworkError, QueryError } from '../errors';
 
-export class DbtRpcClient {
-    serverUrl: string;
-
-    dbtUnreachableErrorMessage: string;
+export const DEFAULT_HEADERS: Record<string, string> = {
+    'Content-Type': 'application/json',
+};
 
-    constructor(serverUrl: string) {
-        this.serverUrl = serverUrl;
-        this.dbtUnreachableErrorMessage = `Cannot connect to dbt rpc server at ${serverUrl}. Are you sure the process is running correctly?`;
-    }
-
-    private async _postDbtSyncRpc(method: string, params: Object) {
-        const requestId = uuidv4();
-        const payload = {
-            method,
-            params,
-            jsonrpc: '2.0',
-            id: requestId,
-        };
-        const headers = {
-            'Content-Type': 'application-json',
-        };
-        let data: any = {};
-        try {
-            const response = await fetch(this.serverUrl, {
-                method: 'POST',
-                headers,
-                body: JSON.stringify(payload),
-            });
-            data = await response.json();
-        } catch (e) {
-            throw new NetworkError(this.dbtUnreachableErrorMessage, {});
-        }
-        if (data === undefined)
-            throw new NetworkError(this.dbtUnreachableErrorMessage, {});
-        else if (data.error)
-            throw new DbtError(
-                `Error returned from dbt while executing method '${method}': ${
-                    data.error.data.message || data.error.data
-                }`,
-                data.error.data,
-            );
-        else if (data.result?.error)
-            throw new DbtError(
-                `Error returned from dbt while executing method '${method}'`,
-                data.result.error,
-            );
-        else if (
-            data.result?.logs?.some(
-                (log: any) => log.message === 'Please log into GCP to continue',
-            )
-        ) {
-            throw new DbtError(
-                `Lightdash cannot connect to dbt because of missing GCP credentials. This error happened because your profiles.yml contains a bigquery profile with method: oauth`,
-                {},
-            );
-        }
-        return data;
-    }
+export abstract class DbtRpcClientBase {
+    abstract _postDbtSyncRpc(
```

Comment:
If this returns a promise, shouldn't we call it async ? we might need to rethink this name since we already have a `_postDbtAsyncRpc`

Reply:
Yep that's really confusing, it's meant to refer to how the procedure runs over http not a reference to the js function. But I'll change this